src/sentry/integrations/slack/views/link_team.py

```diff
@@ -69,23 +78,36 @@ class SlackLinkTeamView(BaseView):
     Django view for linking team to slack channel. Creates an entry on ExternalActor table.
     """
 
+    _METRICS_SUCCESS_KEY = SLACK_BOT_COMMAND_LINK_TEAM_SUCCESS_DATADOG_METRIC
+    _METRICS_FAILURE_KEY = SLACK_BOT_COMMAND_LINK_TEAM_FAILURE_DATADOG_METRIC
+
     @method_decorator(never_cache)
     def handle(self, request: Request, signed_params: str) -> HttpResponseBase:
         if request.method not in ALLOWED_METHODS:
             return render_error_page(request, status=405, body_text="HTTP 405: Method not allowed")
 
         try:
-            params = unsign(signed_params)
-        except (SignatureExpired, BadSignature):
+            converted = unsign(signed_params)
+            params = TeamLinkRequest(**converted)
+        except (SignatureExpired, BadSignature) as e:
+            _logger.warning("handle.signature_error", exc_info=e)
+            metrics.incr(self._METRICS_FAILURE_KEY, tags={"error": str(e)}, sample_rate=1.0)
             return render_to_response(
                 "sentry/integrations/slack/expired-link.html",
                 status=400,
                 request=request,
             )
 
-        integration = integration_service.get_integration(integration_id=params["integration_id"])
+        logger_params = asdict(params)
+        logger_params["user_id"] = request.user.id
+
+        integration = integration_service.get_integration(integration_id=params.integration_id)
         if integration is None:
-            raise Http404
+            _logger.info("integration.not_found", extra=logger_params)
+            metrics.incr(self._METRICS_FAILURE_KEY + ".get_integration", sample_rate=1.0)
+            raise render_error_page(
```

Comment:
should this be `return`? do we want to know about Http404s?

Reply:
1. yep should be return. 
2. we will still return a 404, but instead erroring and not returning anything to the user and leaving them with no context on what went wrong, we instead show them an error page. the logging and metric should give us info about the 404s anyway.